index-import.cpp

```diff
@@ -98,10 +98,27 @@ struct ModuleNameScope {
   std::set<StringRef> _moduleNames;
 };
 
-static IndexUnitWriter remapUnit(const std::unique_ptr<IndexUnitReader> &reader,
-                                 const Remapper &remapper, FileManager &fileMgr,
-                                 ModuleNameScope &moduleNames,
-                                 std::ostream *outs) {
+// Returns true if the Unit file of given output file already exists and is
+// older than the input file.
+static bool isUnitUpToDate(StringRef outputFile, StringRef inputFile,
+                           IndexUnitWriter &writer) {
+  std::string error;
+  auto isUpToDateOpt =
+      writer.isUnitUpToDateForOutputFile(outputFile, inputFile, error);
+  if (!isUpToDateOpt.hasValue()) {
+    errs() << "error: failed file status check:\n" << error << "\n";
```

Comment:
Do you know which cases cause this to return nil? I'm wondering if this is an error, or if it could end up being noise?

Reply:
The code is [here](https://github.com/apple/swift-clang/blob/b4c4d68d76e1432032a930efade53a406e1a3c70/lib/Index/IndexUnitWriter.cpp#L219)

It fails when the unit file does exist but can not be accessed. In particular, when `EC != llvm::errc::no_such_file_or_directory`

Reply:
got it, thanks